### PR TITLE
Fix AssertClose bug in Imgcodec tests

### PR DESCRIPTION
**Pull request by szkarpinski**
## Category:
**Bug fix** (*non-breaking change which fixes an issue*)

## Description:

There were three versions of `AssertClose` in Imgcodec's decoder test helper:
https://github.com/NVIDIA/DALI/blob/edbac3ce6875fee4a405a252e802ce9cb9d2c274/dali/imgcodec/decoders/decoder_test_helper.h#L172-L194

Only one of those interpreted the `eps` as being in dynamic range of the output type (and that's what the doc promised). As a result, `eps=1`, which should mean "we allow +/-1 error in `uint8` image" was interpreted as "we allow +/-1.0 error in **float** image". And, as floats have dynamic range 0.0-1.0, this effectively meant "we allow **any** error".

In this PR I fix the behaviour of the buggy overloads.

The fix is simply making those two overloads call the correct versions of `AssertClose` instead of calling `Check` directly.

## Additional information:

This required adjusting some tests:
- In LibJpegTurbo adjusting the eps was required
- In NvJpeg fixing `AssertClose` revealed another problem: reference images were generated using libjpegturbo, resulting in high (but local) differences. I created a new method `AssertSimilar`, which has a very liberal `eps` (=30%), but also requires the mean error to be relatively small (<1%).

### Tests:

- [x] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [x] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Review by klecki — APPROVED** [review]

File: dali/imgcodec/decoders/nvjpeg/nvjpeg_test.cc [hunk 1]
```diff
@@ -72,6 +72,32 @@ class NvJpegDecoderTest : public NumpyDecoderTestBase<GPUBackend, OutputType> {
     return std::make_shared<JpegParser>();
   }
 
+  /**
+   * @brief Checks if the image is similar to the reference.
+   *
+   * Checks if 1) max error is lower than 30% 2) mean squared error is lower than 1%.
```
Comment (mzient) [inline dali/imgcodec/decoders/nvjpeg/nvjpeg_test.cc:78]: The implementation has mean absolute value (L1 norm), not mean square (L2 norm).

File: dali/imgcodec/decoders/nvjpeg/nvjpeg_test.cc [hunk 2]
```diff
@@ -72,6 +72,32 @@ class NvJpegDecoderTest : public NumpyDecoderTestBase<GPUBackend, OutputType> {
...
+      float eps = ConvertSatNorm<OutputType>(0.3);
+      Check(img, ref, EqualConvertNorm(eps));
+
+      double mean_error = 0;
+      uint64_t size = volume(img.shape);
+      for (size_t i = 0; i < size; i++) {
+        double img_value = ConvertSatNorm<double>(img.data[i]);
+        double ref_value = ConvertSatNorm<double>(ref.data[i]);
+        double error = abs(img_value - ref_value);
+        mean_error += error;
```
Comment (mzient) [inline dali/imgcodec/decoders/nvjpeg/nvjpeg_test.cc:94]: Mean square, as mentioned?
```suggestion
        double error = img_value - ref_value;
        mean_error += error * error;
```

File: dali/imgcodec/decoders/nvjpeg/nvjpeg_test.cc [hunk 3]
```diff
@@ -72,6 +72,32 @@ class NvJpegDecoderTest : public NumpyDecoderTestBase<GPUBackend, OutputType> {
...
+      }
+      mean_error /= size;
```
Comment (mzient) [inline dali/imgcodec/decoders/nvjpeg/nvjpeg_test.cc:96]:
```suggestion
      mean_error = sqrt(mean_error / size);
```

File: dali/imgcodec/decoders/nvjpeg/nvjpeg_test.cc (hunk 1 above)
Comment (szkarpinski) [reply dali/imgcodec/decoders/nvjpeg/nvjpeg_test.cc:78]: Oh, I forgot to change the comment. But I'll make it mean square in a moment as suggested

Comment (szkarpinski) [reply dali/imgcodec/decoders/nvjpeg/nvjpeg_test.cc]: Changed. Thanks for the catch

**Comment by szkarpinski** [comment]
!build

**Review by mzient — APPROVED** [review]

**Pull request merged**